lib/mix/tasks/tic_tac_toe.ex

```diff
@@ -1,7 +1,10 @@
 defmodule Mix.Tasks.TicTacToe do
   use Mix.Task
+  alias TicTacToe.Board
+  alias TicTacToe.Display
+  alias TicTacToe.Presenter
 
   def run(_) do
-    TicTacToe.run()
+    TicTacToe.run(Board, Display, Presenter)
```

Comment:
I think there are times when it makes sense to pass around modules. That can be a good strategy at times. However, I think it is important for us to not think of modules as a 1-1 relationship to classes. (I have personally taken this approach too far and it was very difficult to manage at a certain point. I can see this getting difficult once you have more than one or two layers of abstraction.)

One thing I don't see is some sort of state or config datastructure that we are passing around.  I see that  `empty_grid` is being called inside `.run` but this kind of feels akin to calling `.new` on a class.

I would prefer to see us also leverage protocols where it makes sense. Maybe you could share some context on your decision to inject modules over using some protocols as well?

Reply:
I'm passing around the modules because I thought you can't call a method as such, Display.output, within a function (one of the constraints)?  I didn't pass around a state data structure yet because I didn't need it for this iteration.  I didn't use protocols because I think of them as polymorphic tools - appropriate when we want different objects to respond to the same message.  I don't think we have that situation yet, unless we consider protocols for testing?

Reply:
Cool thanks for that context!

>I thought you can't call a method as such, Display.output, within a function (one of the constraints)?
This is true 😄 

> I didn't pass around a state data structure yet because I didn't need it for this iteration
👍 

> I didn't use protocols because I think of them as polymorphic tools - appropriate when we want different objects to respond to the same message
I think we are in that situation already, right? We need to provide different versions in test.

Reply:
Also though, let's keep in mind that modules are not objects. They are just a namespaced set of functions.

Reply:
it is not an easy switch to make 😄

Reply:
If Display is a protocol with an impl, can I call it as such Display.output in a function?  If so, then worth it to create protocols now because I can see the run argument list growing large.

Reply:
Correct! It is totally ok to have a dependency on a protocol.